### PR TITLE
[Dont merge] 204 : Added suport NX for SET command

### PR DESCRIPTION
**Pull request by chanducheryala**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -14,7 +14,7 @@ import (
 
 func setupFlags() {
 	flag.StringVar(&config.Host, "host", "0.0.0.0", "host for the dice server")
-	flag.IntVar(&config.Port, "port", 7379, "port for the dice server")
+	flag.IntVar(&config.Port, "port", 6379, "port for the dice server")
```
Comment (arpitbbhayani) [inline main.go]: Do not change the port. The default port of DiceDB is 7379.

File: core/eval.go [hunk 2]
```diff
@@ -87,6 +87,11 @@ func evalSET(args []string) []byte {
 				return Encode(errors.New("ERR value is not an integer or out of range"), false)
 			}
 			exDurationMs = exDurationSec * 1000
+		case "NX", "nx":
+			obj := Get(key);
+			if obj != nil {
+				return Encode(fmt.Errorf("ERR %s key exists", key), false);
```
Comment (arpitbbhayani) [inline core/eval.go:93]: Redis does not raise an error if the object already exists. It simply ignores. it. Stick to how Redis behaves under similar situation.

File: tests/set_test.go [hunk 3]
```diff
@@ -21,3 +21,45 @@ func TestSet(t *testing.T) {
...
+func TestSetWithNX(t *testing.T) {
+	conn := getLocalConnection()
+	for _, tcase := range []DTestCase{
+		{
+			InCmds: []string{"SET K V NX", "GET K"},
+			Out:    []interface{}{"OK", "V"},
+		},
+		{
+			InCmds: []string{"SET k v NX", "SET k v NX"},
+			Out:    []interface{}{"OK", "ERR k key exists"},
```
Comment (arpitbbhayani) [inline tests/set_test.go:33]: This is inappropriate case.

File: tests/set_test.go [hunk 4]
```diff
@@ -21,3 +21,45 @@ func TestSet(t *testing.T) {
...
+func BenchmarkSetWithNX(b *testing.B) {
+	conn := getLocalConnection()
+	for _, tcase := range []DTestCase{
+		{
+			InCmds: []string{"SET K V NX", "GET K"},
+			Out:    []interface{}{"OK", "V"},
+		},
+		{
+			InCmds: []string{"SET k v NX", "SET k v NX"},
+			Out:    []interface{}{"OK", "ERR k key exists"},
```
Comment (arpitbbhayani) [inline tests/set_test.go:53]: Should not be a error.

**Review by arpitbbhayani — CHANGES_REQUESTED** [review]

File: main.go (hunk 1 above)
Comment (chanducheryala) [reply main.go]: the file was added mistakely, sorry for that

**Pull request closed without merging**